src/kontrol/kdist/cheatcodes.md

```diff
@@ -345,6 +345,7 @@ The `<output>` cell will be updated with the value of the address generated from
          <k> #call_foundry SELECTOR ARGS => . ... </k>
          <output> _ => #bufStrict(32, #addrFromPrivateKey(#unparseDataBytes(ARGS))) </output>
       requires SELECTOR ==Int selector ( "addr(uint256)" )
+      [preserves-definedness]
```

Comment:
In this rule, the construction of `#addrFromPrivateKey` ensures that the resulting value is in the range `[0, pow256)`.

Reply:
Is `#unparseDataBytes(ARGS)` total? From looking at its defining rule, I have some doubts.

Reply:
We have that:
```
rule #unparseDataBytes( DATA ) => replaceFirst(Base2String(#asInteger(#asByteStack(1) +Bytes DATA), 16), "1", "0x")
```
Then, since `#asByteStack`, `+Bytes`, `#asInteger`, and `replaceFirst` are total by definition, the only possible issue would be that `Base2String` could be undefined for its arguments. From the documentation we have:
```
...
You can also convert between a `String` and an `Int` in a particular radix.
This radix can be anywhere between 2 and 36. For a radix 2 <= N <= 10, the
digits 0 to N-1 will be used. For a radix 11 <= N <= 36, the digits 0 to 9
and the first N-10 letters of the Latin alphabet will be used.
...
```
and so the partiality of `Base2String` seems to be in the second argument, which needs to be between 2 and 36, which holds, since it is 16. Another possible partiality would be that the first argument cannot be less than zero, but it isn't since `#asInteger` always returns a non-negative number. Therefore, I believe that `#unparseDataBytes` is effectively total.